test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -88,4 +92,50 @@ static public void testPassingNullToIsFlattenedThrowsException() throws Throwabl
 			myUnsafe.isFlattened(null);
 		});
 	}
+
+	@Test
+	static public void testValueHeaderSizeOfValueType() throws Throwable {
+		List<String> arguments = ManagementFactory.getRuntimeMXBean().getInputArguments();
+		if (arguments.contains("-Xnocompressedrefs")) {
+			assertEquals(myUnsafe.valueHeaderSize(ValueTypePoint2D.class), 8);
+		} else {
+			assertEquals(myUnsafe.valueHeaderSize(ValueTypePoint2D.class), 4);
+		}
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfPaddedValueType() throws Throwable {
+		assertEquals(myUnsafe.valueHeaderSize(ThisClassWillHavePaddingWhenReferencesAreCompressed.class), 8);
+	}
+
+	// TODO (#14117): add testing for Unsafe.valueHeaderSize using Unsafe.getObjectSize
+
+	@Test
+	static public void testValueHeaderSizeOfNonValueType() throws Throwable {
+	    assertEquals(myUnsafe.valueHeaderSize(Point2D.class), 0);
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfNull() throws Throwable {
+	    assertEquals(myUnsafe.valueHeaderSize(null), 0);
+	}
+
```

Comment:
I think there should be a test case that `myUnsafe.uninitializedDefaultValue()` returns NULL on a value type.

Reply:
addressed in 780435c